code_generator/GdkCodeGenerator/Templates/UnityGameObjectComponentDispatcherGenerator.tt

```diff
@@ -192,16 +234,80 @@ namespace <#= qualifiedNamespace #>
                         continue;
                     }
 
-                    var authChanges = authChangeLists[i];
-                    foreach (Requirables.ReaderWriterImpl readerWriter in readersWriters)
+                    if (isFirstAuthChange(Authority.Authoritative, changeOpsLists[i]))
                     {
-                        foreach (var auth in authChanges.Changes)
+                        foreach (Requirables.ReaderWriterImpl readerWriter in readersWriters)
                         {
-                            readerWriter.OnAuthorityChange(auth);
+                            readerWriter.OnAuthorityChange(Authority.Authoritative);
                         }
                     }
                 }
             }
+
+            public override void InvokeOnAuthorityLostCallbacks(Dictionary<int, InjectableStore> entityIndexToInjectableStore)
+            {
+                if (AuthorityLostComponentGroup.IsEmptyIgnoreFilter)
+                {
+                    return;
+                }
+
+                var entities = AuthorityLostComponentGroup.GetEntityArray();
+                var changeOpsLists = AuthorityLostComponentGroup.GetComponentDataArray<AuthorityChanges<<#= componentDetails.TypeName #>>>();
+
+                // Call once on all entities unless they flip-flopped back into the state they started in
+                for (var i = 0; i < entities.Length; i++)
+                {
+                    var injectableStore = entityIndexToInjectableStore[entities[i].Index];
+                    if (!injectableStore.TryGetInjectablesForComponent(readerWriterInjectableId, out var readersWriters))
+                    {
+                        continue;
+                    }
+
+                    if (isFirstAuthChange(Authority.NotAuthoritative, changeOpsLists[i]))
+                    {
+                        foreach (Requirables.ReaderWriterImpl readerWriter in readersWriters)
+                        {
+                            readerWriter.OnAuthorityChange(Authority.NotAuthoritative);
+                        }
+                    }
+                }
+            }
+
+            private bool isFirstAuthChange(Authority authToMatch, AuthorityChanges<<#= componentDetails.TypeName #>> changeOps)
```

Comment:
Style: Upper-case to "IsFirstAuthChange"
This whole ignoring of multiple auth changes in a frame is dubious to me, however.